agent/xds/server.go

```diff
@@ -263,10 +263,9 @@ func (s *Server) process(stream ADSStream, reqCh <-chan *envoy.DiscoveryRequest)
 				return status.Errorf(codes.PermissionDenied, "permission denied")
 			}
 		case structs.ServiceKindMeshGateway:
```

Comment:
trivial: you could remove the use of 2 `fallthrough` statements here if you made this refactor:
```
switch v {
  case a:
    fallthrough
  case b:
    fallthrough
  case c:
    // body
}
```
to
```
switch v {
  case a, b, c:
    // body
}
```

That one might be less accident prone in the future.

Reply:
True, just changed it.